### PR TITLE
Request a list of tasks with params

### PR DESCRIPTION
**Pull request by phacops**
This new method allows to send a request with different params to iron.io's API in order to filter what is returned.

### CODE REVIEW

File: worker/methods.go [hunk 1]
```diff
@@ -238,6 +238,52 @@ func (w *Worker) TaskList() (tasks []TaskInfo, err error) {
...
+	FromTime time.Time
+	ToTime   time.Time
+	Statuses []string
+}
+
+func (w *Worker) FilteredTaskList(params TaskListParams) (tasks []TaskInfo, err error) {
+	out := map[string][]TaskInfo{}
+	url := w.tasks()
+
+	if params.CodeName != "" {
```
Comment (featalion) [inline worker/methods.go]: This looks like `code_name` parameter, represented by `TaskListParams.CodeName`, is optional API parameter. Could you please remove the check, that `CodeName` is empty, and keep just `url.QueryAdd(...)`.

Comment (phacops) [inline worker/methods.go]: I just don't want an empty parameter to be included in the request. That's why I check for emptyness. The request will be fired even if it's empty. I can still remove it though.

By the way, according to the documentation (http://dev.iron.io/worker/reference/api/#list_tasks), it's a required parameter.

Comment (featalion) [inline worker/methods.go]: > I just don't want an empty parameter to be included in the request.

what's wrong with empty parameters in request? It is regular case for HTTP. Let worker's API handle all the errors ( ;

> it's a required parameter

exactly. But in the client library it looks like optional.

Comment (phacops) [inline worker/methods.go]: True.

**Comment by featalion** [comment]
Thanks, @phacops !

**Pull request merged**